### PR TITLE
Navigation component

### PR DESCRIPTION
**Pull request by Efekahya**
image
https://www.figma.com/file/SBPCNjmePIwJCzqqhXF9Y8/Rick-and-Morty---Digieggs-Assessment?node-id=1019%3A10308

### CODE REVIEW

File: src/components/Navigation/index.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      {href.map((item, index) => {
+        for (let i = 0; i < href.length; i++) {
+          if (href[i] === item) {
+            if (i === 0) {
+              temp = href[0]
+            } else {
+              temp = href.slice(0, i + 1).join("/")
+            }
+          }
+        }
```
Comment (bybash) [inline src/components/Navigation/index.tsx]: Make these operations before render function.

File: src/App.tsx [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+
 function App() {
   return (
     <div className="App">
-      <Navbar Logo={<Logo />} />
+      <BrowserRouter>
+        <Routes>
+          <Route path="*" element={<Navigation />} />
+        </Routes>
+      </BrowserRouter>
```
Comment (bybash) [inline src/App.tsx]: You don't need to send this kind of change in your PRs. Make your tests in your local and don't push them inside the PRs.

**Review by bybash — CHANGES_REQUESTED** [review]
👍 noted down some comments.

File: src/components/Navigation/index.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
```
Comment (bybash) [inline src/components/Navigation/index.tsx]: Create an array of URLs which you will give to the `<a>` tag before the render method and then just render.
```suggestion
var urlsArray = href.map((item) => {
         for (let i = 0; i < href.length; i++) {
           if (href[i] === item) {
             if (i === 0) {
               temp = href[0]
             } else {
               temp = href.slice(0, i + 1).join("/")
             }
           }
         }

      {urlsArray.map((item, index) => (
           <>
             <div key={index} className="navigationText">
               <a href={"/" + item}>{hrefUpper[index]}</a>
             </div>
             <RightArrow />
           </>
        )
       }
```

**Review by bybash — CHANGES_REQUESTED** [review]

File: src/components/Navigation/index.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+let temp: string
+
+const urlsArray = href.map((url, index) => {
+  if (index === 0) {
+    temp = url
+    return url
+  } else {
+    temp += "/" + url
```
Comment (atvdi) [inline src/components/Navigation/index.tsx]:
```suggestion
    temp += `/ ${url}`
```

File: src/components/Navigation/index.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    return temp
+  }
+})
+
+const elements = urlsArray.map((url, index) => (
+  <>
+    <div key={index} className="navigation--navigationText">
+      <a href={"/" + url}>{hrefUpper[index]}</a>
```
Comment (atvdi) [inline src/components/Navigation/index.tsx]:
```suggestion
      <a href={`/ ${url}`}>{hrefUpper[index]}</a>
```

File: src/components/Navigation/styles.scss [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+.navigation--navigationText {
+  @include interText(700, 16px, 19px, $black);
+
+  a {
+    text-decoration: none;
+    color: $black;
+  }
+
+  a:hover {
+    color: #808080;
```
Comment (atvdi) [inline src/components/Navigation/styles.scss]: I recommend variable here.

**Review by atvdi — CHANGES_REQUESTED** [review]

**Pull request merged**